e2e/scenario_test.go

```diff
@@ -901,3 +904,33 @@ func Test_ubuntu2204Wasm(t *testing.T) {
 		},
 	})
 }
+
+func Test_noEchoedSecretsInProvisionLog(t *testing.T) {
+	logPath := "/var/log/azure/cluster-provision.log"
+	secretContent := "**** this secret should not be logged ****"
+	clientPrivateKey := base64.StdEncoding.EncodeToString([]byte("ClientPrivateKey: " + secretContent))
+	spSecret := base64.StdEncoding.EncodeToString([]byte("ServicePrincipalSecret: " + secretContent))
+
+	// Getting the cluster so we can also check for the bootstrap token.
+	cluster, err := ClusterKubenet(newTestCtx(t), t)
+	require.NoError(t, err)
+
+	bootstrapToken := *cluster.NodeBootstrappingConfiguration.KubeletClientTLSBootstrapToken
+
+	RunScenario(t, &Scenario{
+		Description: "tests that none of this set of secrets have been echoed into the provision log",
+		Config: Config{
+			Cluster: ClusterKubenet,
+			VHD:     config.VHDUbuntu2204Gen2Containerd,
+			BootstrapConfigMutator: func(nbc *datamodel.NodeBootstrappingConfiguration) {
+				nbc.ContainerService.Properties.CertificateProfile.ClientPrivateKey = clientPrivateKey
+				nbc.ContainerService.Properties.ServicePrincipalProfile.Secret = spSecret
+			},
+			LiveVMValidators: []*LiveVMValidator{
+				FileExcludesContentsValidator(logPath, clientPrivateKey, "client private key"),
```

Comment:
I'd probably opt for having these added to the set of common live VM validators we execute in _all_ secnarios, rather than having a specific scenario for this purpose